index.js

```diff
@@ -92,36 +99,30 @@ function createThreadInterceptor (opts) {
 
         if (err) {
           hooks.fireOnClientError(newOpts, res, clientCtx, err)
-          handler.onError(err)
+          handler.onResponseError(controller, err)
           return
         }
         hooks.fireOnClientResponse(newOpts, res, clientCtx)
 
-        const headers = []
-        for (const [key, value] of Object.entries(res.headers)) {
-          if (Array.isArray(value)) {
-            for (const v of value) {
-              headers.push(key)
-              headers.push(v)
-            }
-          } else {
-            headers.push(key)
-            headers.push(value)
+        try {
+          handler.onRequestStart(controller, {})
+          if (controller.aborted) {
+            handler.onResponseError(controller, controller.reason)
+            return
           }
+          handler.onResponseStart(
```

Comment:
Can the response be aborthed in the `onResponseStart` (I think so)? Do we need a `controller.aborted` chech after this call?

Reply:
I've added the lines and skipped the tests.